BitbloqEvolution/BitbloqEvolution.cpp

```diff
@@ -0,0 +1,285 @@
+}
+
+bool Evolution::getRestHeadState(){
+
+    return isEvolutionHeadResting;
+}
+
+void Evolution::setRestHeadState(bool state){
+
+    isEvolutionHeadResting = state;
+}
+
+
+///////////////////////////////////////////////////////////////////
+//-- PREDETERMINED MOTIONS  -------------------------------------//
+///////////////////////////////////////////////////////////////////
+
+void Evolution::stop(){
+
+  move(0, 0);
+}
+
+void Evolution::fordward(int V){
+
+  move(V, V);
+}
+
+void Evolution::backward(int V){
+
+  move(-V, -V);
+}
+
+void Evolution::right(int V){
+
+  move(V, 0);
+}
+
+void Evolution::left(int V){
+
+  move(0, V);
+}
+
+
+///////////////////////////////////////////////////////////////////
+//-- SENSORS FUNCTIONS  -----------------------------------------//
+///////////////////////////////////////////////////////////////////
+
+//-----------------------------------------------------------------------
+//-- Evolution getDistance: return Evolution's ultrasonic sensor measure
+//-----------------------------------------------------------------------
+float Evolution::getDistance(){
+
+  return us.read();
+}
+
+//-----------------------------------------------------------------------
+//-- Evolution getLight: return Evolution's LDR sensor measure
+//-----------------------------------------------------------------------
+int Evolution::getLight(int side){
+
+  if (side > 0){
+    return analogRead(PIN_LDRL);
+  }
+  return analogRead(PIN_LDRR);
+}
+
+//-----------------------------------------------------------------------
+//-- Evolution getLine: return Evolution's LDR sensor measure
+//-----------------------------------------------------------------------
+int Evolution::getLine(int side){
+
+  if (side > 0){
+    //return _linefollower.readLeftSensor();
+    return digitalRead(PIN_LFL);
+  }
+  //return _line_follower.readRightSensor();
+  return digitalRead(PIN_LFR);
+}
+
+bool Evolution::getLightRange(int side, int range){
+
+  int read;
+
+  if (side > 0){
+    read = analogRead(PIN_LDRL);
+  }
+  else {
+    read = analogRead(PIN_LDRR);
+  }
+
+  if (read > MEDIUM_HIGH_LIGHT && range == HIGH_LIGHT){
+    return true;
+  }
+  else if (read <= MEDIUM_HIGH_LIGHT && read >= LOW_MEDIUM_LIGHT && range == MEDIUM_LIGHT){
+    return true;
+  }
+  else if (read < LOW_MEDIUM_LIGHT && range == LOW_LIGHT) {
+    return true;
+  }
+
+  return false;
+}
+
+///////////////////////////////////////////////////////////////////
+//-- SOUNDS -----------------------------------------------------//
+///////////////////////////////////////////////////////////////////
+
+void Evolution::_tone (float noteFrequency, long noteDuration, int silentDuration){
+
+      if(silentDuration==0){silentDuration=1;}
+
+      tone(pinBuzzer, noteFrequency, noteDuration);
+      delay(silentDuration);     
+}
+
+
+void Evolution::bendTones (float initFrequency, float finalFrequency, float prop, long noteDuration, int silentDuration){
```

Comment:
This function does not make sense here. It is only used to create Zowi's language.